### PR TITLE
Update table content validation for large tables

### LINKED ISSUE #533: Bug in validate when reading products with a number of records > 4-byte range
## 🐛 Describe the bug

From user:

I think I have found an issue in validate (version 2.3.0). I am working on migrating GRAIL gravity models to PDS4. Some of these products are very big because of the high degree and order of the models. In particular, I got validation errors for some of the larger binary models, each of which consist of 4 tables. The fourth table is a covariance matrix table that is modeled as a table with one column and many rows in PDS3. I kept that approach for the PDS4 label.

Attached is the validation report and an example label. I omit the data file because products with reported errors range in size from 122 GBytes to 2.5 TBytes. Ignore the noted error about missing collections because I have not finished creating those collections.

GRAIL binary spherical harmonic models (SHBDR) have covariance tables with up to 329,507,590,503 (3.29x10+11) records. These values exceed the range of a 4-byte unsigned integer and probably is the cause of the error message from validate. However, the current IM shows a maximum value for records in a table to be 18,446,744,073,709,551,615 (~ 1.8x10+19), which implies it should be treated as a 8-byte unsigned integer.

## 📜 To Reproduce
Run validate against test label.

## 📚 Version of Software Used

2.3.0

## 🩺 Test Data / Additional context

[Archive.zip](https://github.com/NASA-PDS/validate/files/9366249/Archive.zip)
[val03a_con.txt](https://github.com/NASA-PDS/validate/files/9366250/val03a_con.txt)

## 🖥 System Info

---

### PR DESCRIPTION
**Pull request by jordanpadams**
## 🗒️ Summary
See commit messages in PR for details.

NOTE: This fixes #533 by setting any references to record counts to be a `long` instead of an `int`. Additionally, it includes at start at some code cleanup to hopefully help the code make slightly more sense and fixed additional bugs that popped up in the code execution, but still needs a lot of work.

## ⚙️ Test Data and/or Report
See 0537dbb55d49120d81e73a48ae33003dd7882710 and 6c0906bdbbc148f0f5a661ca85b33e51daab817b.

NOTE: Github Actions tests will fail. These updates require install updates from [this PDS4 JParser PR](https://github.com/NASA-PDS/pds4-jparser/pull/58) prior to running these tests.

```
# In cloned pds4-jparser repo
git checkout val533
mvn clean install

# Once that has completed successfully
# From validate repo
git checkout i533
mvn test
```

## ♻️ Related Issues

Fixes #533

### CODE REVIEW

File: src/test/java/cucumber/StepDefs.java [hunk 1]
```diff
@@ -226,6 +226,10 @@ private int getMessageCountBasedOnProblemType(String problemEnum, JsonObject rep
                 count = this.getMessageCount(reportJson, ProblemType.ARRAY_VALUE_OUT_OF_MIN_MAX_RANGE.getKey());
             } else if (strTemp.equals("NO_PRODUCTS_FOUND")) {
                 count = this.getMessageCount(reportJson, ProblemType.NO_PRODUCTS_FOUND.getKey());
+            } else if (strTemp.equals("TABLE_FILE_READ_ERROR")) {
+                count = this.getMessageCount(reportJson, ProblemType.TABLE_FILE_READ_ERROR.getKey());
+            } else if (strTemp.equals("FIELD_VALUE_OUT_OF_MIN_MAX_RANGE")) {
+                count = this.getMessageCount(reportJson, ProblemType.FIELD_VALUE_OUT_OF_MIN_MAX_RANGE.getKey());
```
Comment (nutjob4life) [inline src/test/java/cucumber/StepDefs.java:232]: 🤯  this `getMessageCountBasedOnProblemType` is the _opposite_ of object oriented code omg my eyes 👀 😀

File: src/main/java/gov/nasa/pds/tools/validate/content/table/TableContentProblem.java [hunk 2]
```diff
@@ -130,7 +154,7 @@ public Integer getTable() {
    * 
    * @return the record index.
    */
-  public Integer getRecord() {
+  public long getRecord() {
```
Comment (nutjob4life) [inline src/main/java/gov/nasa/pds/tools/validate/content/table/TableContentProblem.java:157]: What's the reason going from object `Integer` to primitive `long` and not object `Long`?

File: src/test/java/cucumber/StepDefs.java (hunk 1 above)
Comment (nutjob4life) [inline src/test/java/cucumber/StepDefs.java:232]: This `getMessageCountBasedOnProblemType`function omg my eyes 👀 😆

Maybe use introspection here? The reflection API could turn this 113 line function into 8–10 lines 🤔

File: src/main/java/gov/nasa/pds/tools/validate/content/table/TableContentProblem.java [hunk 3]
```diff
@@ -32,7 +32,7 @@ public class TableContentProblem extends ContentProblem {
   private Integer table;
   
   /** The record number associated with the problem. */
-  private Integer record;
+  private long record;
```
Comment (nutjob4life) [inline src/main/java/gov/nasa/pds/tools/validate/content/table/TableContentProblem.java:35]: What's the reason going from object Integer to primitive long and not object Long?

**Review by nutjob4life — CHANGES_REQUESTED** [review]
I have some questions

File: src/main/java/gov/nasa/pds/tools/validate/content/table/TableContentProblem.java (hunk 2 above)
Comment (jordanpadams) [reply src/main/java/gov/nasa/pds/tools/validate/content/table/TableContentProblem.java:157]: @nutjob4life nothing in particular except I don't know the original rationale for it being `Integer` to begin with?

Comment (jordanpadams) [reply src/main/java/gov/nasa/pds/tools/validate/content/table/TableContentProblem.java:157]: I don't feel strongly either way.

File: src/main/java/gov/nasa/pds/tools/validate/content/table/TableContentProblem.java (hunk 3 above)
Comment (jordanpadams) [reply src/main/java/gov/nasa/pds/tools/validate/content/table/TableContentProblem.java:35]: See above.

File: src/test/java/cucumber/StepDefs.java (hunk 1 above)
Comment (jordanpadams) [reply src/test/java/cucumber/StepDefs.java:232]: Yeah... It pains me to add stuff to it, but I just ran out of time. I will create a ticket to refactor this.

File: src/main/java/gov/nasa/pds/tools/validate/content/table/TableContentProblem.java (hunk 3 above)
Comment (nutjob4life) [reply src/main/java/gov/nasa/pds/tools/validate/content/table/TableContentProblem.java:35]: Well at least Java does "auto-boxing" so it'll probably never be an issue; let's just ignore it!

File: src/main/java/gov/nasa/pds/tools/validate/content/table/TableContentProblem.java (hunk 2 above)
Comment (nutjob4life) [reply src/main/java/gov/nasa/pds/tools/validate/content/table/TableContentProblem.java:157]: Auto-boxing and un-boxing for the win I guess 😁

File: src/main/java/gov/nasa/pds/tools/validate/content/table/TableContentProblem.java (hunk 3 above)
Comment (nutjob4life) [reply src/main/java/gov/nasa/pds/tools/validate/content/table/TableContentProblem.java:35]: Roger roger

**Review by nutjob4life — APPROVED** [review]
Okay, I'm satisfied now!
Let's light this candle

**Comment by jordanpadams** [comment]
Confirming that after pds4-jparser merge, validate integration tests executed successfully: https://github.com/NASA-PDS/validate/actions/runs/2928390330 

also, new ticket created to refactor `StepDefs.java`: https://github.com/NASA-PDS/validate/issues/537

**Pull request merged**